### PR TITLE
Anomalous Valley Of Tyr

### PR DESCRIPTION
**Pull request by FluffMedic**
Aaaa
## About The Pull Request

Ported most of my tyr project.
Discard some bits, some because I wasn't fond of some bits, and others for a slower approach.

Things should work and be playable.
Aware of one bug, but that only makes some content not accessible as intended.
Unsure or how/when a TM would be good, for more public testing and feedback

Still intend to add further in this PR and/or later PRs, whichever works with energy, time, etc
## Changelog
:cl:
add: The planet of Tyr, an anomalous valley landing spot. Replacement for valley. Beta stage
/:cl:

### CODE REVIEW

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/tyr_base.dm [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	minbodytemp = 175
+	maxbodytemp = 500
+	cold_resist = -0.5
+	heat_resist = 0.75
+
+	see_in_dark = 3 //stealth time?
+
+	icon = 'modular_chomp/icons/mob/tyr.dmi'
+
+	faction = "tyr"
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/tyr_base.dm:13]: factions use defines now

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/ants.dm [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+	layer = HIDING_LAYER
+	health = 3
+	grow_as = list(/mob/living/simple_mob/animal/tyr/mineral_ants,
+	/mob/living/simple_mob/animal/tyr/mineral_ants/bronze,
+	/mob/living/simple_mob/animal/tyr/mineral_ants/copper,
+	/mob/living/simple_mob/animal/tyr/mineral_ants/agate,
+	/mob/living/simple_mob/animal/tyr/mineral_ants/quartz,
+	/mob/living/simple_mob/animal/tyr/mineral_ants/painite,
+	/mob/living/simple_mob/animal/tyr/mineral_ants/builder)
+	faction = "mineral ant"
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/ants.dm:309]: factions use defines now

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/tyr_base.dm [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#define FACTION_TYR "tyr"
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/tyr_base.dm:1]: Goes to defines, not here.

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/heads.dm [hunk 4]
```diff
@@ -366,3 +366,194 @@
 		var/turf/T = pick(bomb_range)
 		P.launch_projectile(target, BP_TORSO, src)
 		bomb_range -= T
+
+/mob/living/simple_mob/mechanical/combat_drone/artillery
+	faction = "eclipse"
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/heads.dm:371]: factions use defines now

**Review by Kashargul — CHANGES_REQUESTED** [review]

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/tyr_base.dm (hunk 1 above)
Comment (FluffMedic) [reply modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/tyr_base.dm]: I was dumb when reading the define thing initially, sorry.

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/ants.dm [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+/obj/effect/ant_structure/webbarrier
+	name = "weblike barrier"
+	icon_state = "web"
+
+/obj/effect/ant_structure/webbarrier/CanPass(atom/movable/mover, turf/target)
+	if(istype(mover, /mob/living/simple_mob/animal/giant_spider))
+		return TRUE
+	else if(istype(mover, /mob/living))
+		if(prob(80))
+			to_chat(mover, span("warning", "You get stuck in \the [src] for a moment."))
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/ants.dm:320]: span_warning("text")

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm [hunk 6]
```diff
@@ -889,3 +784,258 @@
...
+	special_attack_min_range = 1
+	special_attack_max_range = 7
+
+	projectiletype = /obj/item/projectile/energy/spikeenergy_ball //using the weapon found upon tyr
+
+/mob/living/simple_mob/humanoid/eclipse/lunar/experimenter/do_special_attack(atom/A)
+	var/D = src
+	var/mob/living/carbon/human/H = A
+	H.throw_at(D, 2, 4) // Just yoinked.
+	visible_message("<span class='danger'>The [src]'s armor glows silver, pulling \[A] closer!</span>")
```
Comment (Kashargul) [inline modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/eclipse.dm:903]: span_danger("message")

File: modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/ants.dm (hunk 5 above)
Comment (Kashargul) [reply modular_chomp/code/modules/mob/living/simple_mob/subtypes/animal/tyr/ants.dm:320]: It's off with all the span macros used

**Comment by Kashargul** [comment]
One last small question, span_orange will be unsorted in the chat, is that intended or are those meant to be orange coloured infos?

**Comment by FluffMedic** [comment]
Oh those.
I noticed that the normal warning messages weren't noticed by folks, so I was using orange for them to stand out. It's meant to be orange colored information

**Comment by Kashargul** [comment]
> Oh those. I noticed that the normal warning messages weren't noticed by folks, so I was using orange for them to stand out. It's meant to be orange colored information

ok, then best to do span_warning(span_orange("text"))

so that players still can see it in their combat tabs.

**Review by Kashargul — APPROVED** [review]

**Comment by Darlantanis** [comment]
May want zone dividers on the large open areas. If this is entirely simulated turfs, I would expect ZAS to encounter trouble whenever atmos processes

If you want to keep temperature change from weather effects, would need at least one planetary turf per divided zone

**Pull request merged**